controller/CosmeticAPIController.py

```diff
@@ -0,0 +1,254 @@
+import json
+import logging
+import sqlalchemy
+
+from flask import request, jsonify, make_response
+from flask_jwt_extended import jwt_required
+from sqlalchemy import exc
+from controller import app
+from dao import ProductRepo, BrandRepo, CategoryRepo, ColorRepo, ProductTypeRepo, TagRepo
+from dao.models import Product, Brand, Category, Color, ProductType, Tag
+
+
+@app.route("/")
+@jwt_required
+def home():
+    return app.send_static_file('index.html')
+
+
+"""
+API Call to get the product 
+Input : product_id
+"""
+
+
+@app.route('/product/<int:product_id>', methods=['GET'])
+@jwt_required
+def get_product(product_id):
+    try:
+        logging.info(f'getProduct: User has given input product_id as: {product_id}')
+        product = ProductRepo.getProduct(product_id)
+        if product is None:
+            return make_response(jsonify(response=f'Product with id {product_id} not found'), 404)
+        return jsonify(product=Product.serialize(product))
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+API Call to search the product 
+Input: 
+        brand
+        product_category
+        product_type
+        price : greater_than 
+                less_than
+        rating:greater_than
+               less_than
+        tag : String or list separated by commas
+"""
+
+
+@app.route('/searchproduct', methods=['GET'])
```

Comment:
```suggestion
@app.route('/search_product', methods=['GET'])
```
You can use snake casing for the routes